lib/Target/DebugInfo/EmitHGLDD.cpp

```diff
@@ -904,6 +881,30 @@ EmittedExpr FileEmitter::emitExpression(Value value) {
             muxOp.getType()};
   }
 
+  // As a last resort, look for any named wire-like ops this value feeds into.
+  // This is useful for instance output ports for example since we cannot access
+  // instance ports as `<instName>.<portName>` in HGLDD. Instead, we look for a
+  // wire hooked up to the instance output, which is very likely to be present
+  // after Verilog emission.
```

Comment:
Does this need a condition `hasSingleUse()`? I think it might be unpredictable to rely on the order of uses.

Reply:
We do want this for instance ports with multiple uses. Otherwise the instance ports would disappear in the HGLDD output. I think the order of uses is deterministic, but somewhat hard to predict. Since the name we dig up here is not user-facing, we should be fine if the name is a somewhat random but deterministic pick among the available options.